inspirehep/modules/citations/tasks.py

```diff
@@ -0,0 +1,49 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of INSPIRE.
+# Copyright (C) 2015, 2016 CERN.
+#
+# INSPIRE is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License as
+# published by the Free Software Foundation; either version 2 of the
+# License, or (at your option) any later version.
+#
+# INSPIRE is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with INSPIRE; if not, write to the Free Software Foundation, Inc.,
+# 59 Temple Place, Suite 330, Boston, MA 02111-1307, USA.
+
+from invenio_celery import celery
+
+
+@celery.task
+def update_citation_count_for_records(recids):
+
+    for recid in recids:
+        update_citation_count.delay(recid)
+
+
+@celery.task
+def update_citation_count(recid):
+    from invenio_ext.es import es
+    from .utils import calculate_citation_count
+    from elasticsearch import TransportError
+    try:
+        query_result = es.get(index='hep', id=recid)
+    except TransportError:
+        query_result = None
+
+    if query_result:
+        record = query_result.get('_source')
```

Comment:
You can get the source directly in one command with: `es.get_source()`. Additionally, do you actually need any field from `record`? If you are just adding the `citation_count` field, then you can avoid getting the `_source` at all, and just use `es.update` with partial doc: http://elasticsearch-py.readthedocs.org/en/master/api.html#elasticsearch.Elasticsearch.update.

In this way we would save a lot in networking, especially with our giant records :dancer: